src/Controls/src/BindingSourceGen/SourceGenerator/PathParser.cs

```diff
@@ -44,14 +44,20 @@ private Result<List<IPathPart>> HandleMemberAccessExpression(MemberAccessExpress
         var typeInfo = Context.SemanticModel.GetTypeInfo(memberAccess).Type;
         var symbol = Context.SemanticModel.GetSymbolInfo(memberAccess).Symbol;
 
-        if (symbol is IFieldSymbol fieldSymbol && fieldSymbol.DeclaredAccessibility == Accessibility.Private)
+        if (symbol is IFieldSymbol fieldSymbol
+            && fieldSymbol.DeclaredAccessibility != Accessibility.Public
+            && fieldSymbol.DeclaredAccessibility != Accessibility.Internal
+            && fieldSymbol.DeclaredAccessibility != Accessibility.ProtectedOrInternal)
         {
-            return Result<List<IPathPart>>.Failure(DiagnosticsFactory.PrivateFieldInPath(memberAccess.GetLocation()));
+            return Result<List<IPathPart>>.Failure(DiagnosticsFactory.UnaccessibleFieldInPath(memberAccess.GetLocation()));
         }
 
-        if (symbol is IPropertySymbol propertySymbol && propertySymbol.DeclaredAccessibility == Accessibility.Private)
+        if (symbol is IPropertySymbol propertySymbol
+        && propertySymbol.DeclaredAccessibility != Accessibility.Public
+        && propertySymbol.DeclaredAccessibility != Accessibility.Internal
+        && propertySymbol.DeclaredAccessibility != Accessibility.ProtectedOrInternal)
```

Comment:
Nit: please extract this to a method, I see this condition in multiple places so let's make sure the conditions are always in sync.